src/core/analytics.js

```diff
@@ -1148,6 +1188,13 @@ class Analytics {
       this.loadIntegration = !!options.loadIntegration;
     }
 
+    if (options && options.bufferDataPlaneEventsUntilReady != undefined) {
```

Comment:
use type checking in conditions !== or typeOf of Boolean(options.bufferDataPlaneEventsUntilReady)